src/core/ext/transport/chttp2/transport/chttp2_transport.cc

```diff
@@ -1167,6 +1167,7 @@ void grpc_chttp2_add_incoming_goaway(grpc_chttp2_transport* t,
             grpc_core::StatusToString(t->goaway_error).c_str());
   }
   if (t->is_client) {
+    cancel_pings(t, t->goaway_error);
```

Comment:
This doesn't sound correct. Why should we cancel pings on GOAWAYs? There is a possibility that the connection stays alive even after GOAWAYs and we would still want the keepalive ping to be functioning properly

Reply:
note that we are setting connectivity state --> transitive failure later in this function... keep alive pings maybe aren't gonna help us much?

Reply:
That connectivity state is just for upper layers to not send us more RPCs, and to create a new connection if needed. For ongoing RPCs, it'll still help

Reply:
Updated title... original bug fix was the wrong direction. Bug is in the underlying fuzzer not completing the last read after a close.